samples/excel/09-events/selection-changed.yaml

```diff
@@ -0,0 +1,116 @@
+name: Selection Changed
+description: Add and remove an event handler on the selection changed event
+host: EXCEL
+api_set: {}
+script:
+    content: |
+        $("#add-event-handler").click(addEventHandler);
+        $("#remove-last-event-handler").click(removeLastEventHandler);
+        $("#remove-all-event-handlers").click(removeAllEventHandlers);
+
+        let lastHandler: OfficeExtension.EventHandlerResult<Excel.Workbook>;
+        let workbook: Excel.Workbook;
+
+        async function addEventHandler() {
+            try {
+                if (workbook !== undefined) {
+                    await Excel.run(workbook, async (context) => {
+                        workbook = context.workbook;
+                        lastHandler = context.workbook.onSelectionChanged.add(onSelectionChanged)
+
+                        await context.sync();
+                    })
+                } else {
+                    await Excel.run(async (context) => {
+                        workbook = context.workbook;
+                        lastHandler = context.workbook.onSelectionChanged.add(onSelectionChanged)
+
+                        await context.sync();
+                    });
+                }
+                console.log("Event handler added. Try changing the selection.");
+            }
+            catch (error) {
+                OfficeHelpers.Utilities.log(error);
+            }
+        }
+
+        async function removeLastEventHandler() {
+            try {
+                await Excel.run(workbook, async (context) => {
+                    lastHandler.remove();
+
+                    await context.sync();
+
+                    console.log("Last event handler removed");
+                })
+            }
+            catch (error) {
+                OfficeHelpers.Utilities.log(error);
+            }
+        }
+
+        async function removeAllEventHandlers() {
+            try {
+                await Excel.run(workbook, async (context) => {
+                    context.workbook.onSelectionChanged.removeAll();
+
+                    await context.sync();
+
+                    console.log("All event handlers removed");
+                })
+            }
+            catch (error) {
+                OfficeHelpers.Utilities.log(error);
+            }
+        }
+
+        async function onSelectionChanged() {
+            console.log("Selection changed, let's look up the address");
```

Comment:
I didn't literally mean you to have this console.log line :-).  It could probably be merged in to where you actually fetch the address, rather than having two separate lines.  Unless you intentionally wanted this?...